src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/AssetService.cs

```diff
@@ -103,6 +103,14 @@ public bool AttachMeshFilter(GameObject gameObject, AssetId assetId)
             return false;
         }
 
+        public bool AttachDynamicMeshFilter(GameObject gameObject, AssetId assetId, Mesh mesh)
+        {
+            ComponentExtensions.EnsureComponent<MeshRenderer>(gameObject);
+            MeshFilter filter = ComponentExtensions.EnsureComponent<MeshFilter>(gameObject);
+            filter.sharedMesh = mesh;

```

Comment:
This may need to be filter.mesh. It looks like filter.sharedMesh isn't suggested for writing meshes.

Reply:
I find Unity's MeshFilter.mesh/sharedMesh confusing, so I'm not sure. I think the warning about writing is if you modify a mesh using filter.sharedMesh. For example, if you do filter.sharedMesh.vertices = new Vector3[] then you could be inadvertently modifying a mesh that is shared by other objects. In this case, we are not modifying the existing sharedMesh but instead are first creating a Mesh and then assigning that mesh to filter.sharedMesh.

Not sure if I'm making much sense here, but I'm not sure what the right way is to do this or if it even matters which property is used in this case. I expect that assigning to filter.mesh would also be fine.